packable/packable/src/packable/prefix/map.rs

```diff
@@ -0,0 +1,299 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate alloc;
+
+use core::{
+    borrow::Borrow,
+    fmt,
+    hash::Hash,
+    marker::PhantomData,
+    ops::{Deref, DerefMut, Range},
+};
+
+use hashbrown::HashMap;
+
+use crate::{
+    error::UnpackError, map::UnpackMapError, packable::bounded::Bounded, packer::Packer, unpacker::Unpacker, Packable,
+};
+
+/// Wrapper type for `HashMapPrefix<K, V>` with a length prefix.
+/// The set's prefix bounds are provided by `B`, where `B` is a [`Bounded`] type.
+/// The prefix type is the `Bounds` type associated with `B`.
+#[derive(Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[repr(transparent)]
+pub struct HashMapPrefix<K: Eq + Hash, V: PartialEq, B: Bounded> {
+    inner: HashMap<K, V>,
+    bounded: PhantomData<B>,
+}
+
+impl<K: Eq + Hash + fmt::Debug, V: PartialEq + fmt::Debug, B: Bounded> fmt::Debug for HashMapPrefix<K, V, B> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if f.alternate() {
+            write!(f, "{:#?}", self.inner)
+        } else {
+            write!(f, "{:?}", self.inner)
+        }
+    }
+}
+
+impl<K: Eq + Hash, V: PartialEq, B: Bounded> Default for HashMapPrefix<K, V, B> {
+    fn default() -> Self {
+        Self {
+            inner: Default::default(),
+            bounded: PhantomData,
+        }
+    }
+}
+
+impl<K: Eq + Hash, V: PartialEq, B: Bounded> Deref for HashMapPrefix<K, V, B> {
+    type Target = HashMap<K, V>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+/// This is fine as slices cannot be resized.
```

Comment:
Not a slice and can be resized, no?

Reply:
I didn't add this so I'm not 100% sure but yeah that seems problematic. I'll remove but you may have issues when using it, we'll see.